### ISSUE TITLE
Physics server with Multi-Threaded thread model doesn't report error reasons (ERR_EXPLAIN)

### ISSUE THREAD

**Issue by akien-mga** [godotengine/godot#26518]
**Godot version:**
bf6398ae

**OS/device including version:**
Mageia 7 x86_64

**Issue description:**

*Note:* This bug is tested and experienced with the physics thread model, but it might also affect other multi-threaded servers like the renderer.

When using **Multi-Threaded** physics thread model, `ERR_EXPLAIN` error descriptions are not properly shown in the debugger:
![screenshot_20190303_105804](https://user-images.githubusercontent.com/4701338/53693619-3c2c2380-3da3-11e9-9e8b-f6d37b0b37c5.png)

In the terminal, it's hit and miss. Sometimes you also get:
```
ERROR: area_set_shape_disabled: Method/Function Failed.
   At: servers/physics_2d/physics_2d_server_sw.cpp:416
```

And sometimes you do get the expected error (but still only "Method/Function Failed." in the debugger):
```
ERROR: area_set_shape_disabled: Can't change this state while flushing queries. Use call_deferred() or set_deferred() to change monitoring state instead
   At: servers/physics_2d/physics_2d_server_sw.cpp:416.
```

Additionally, the error is only reported once, even if it occurs several times.

When you switch to **Single-Safe** physics thread model, it works fine:
![screenshot_20190303_110053](https://user-images.githubusercontent.com/4701338/53693639-9fb65100-3da3-11e9-9819-20b356ffcbd4.png)

In the terminal:
```
ERROR: area_set_shape_disabled: Can't change this state while flushing queries. Use call_deferred() or set_deferred() to change monitoring state instead
   At: servers/physics_2d/physics_2d_server_sw.cpp:416.
ERROR: area_set_shape_disabled: Can't change this state while flushing queries. Use call_deferred() or set_deferred() to change monitoring state instead
   At: servers/physics_2d/physics_2d_server_sw.cpp:416.
```

**Steps to reproduce:**
- Download MRP
- Press any key to trigger errors when the Area2Ds overlap
- Compare behavior when changing values of `physics/2d/thread_model`.

**Minimal reproduction project:**
[err_explain_debugger.zip](https://github.com/godotengine/godot/files/2922981/err_explain_debugger.zip)

**Comment by akien-mga** [comment]
Still reproducible in the current master branch.

**Comment by KoBeWi** [comment]
Can anyone still reproduce this bug in Godot 3.2.3 or any later release?
There's no error in the MRP.

**Comment by phelioz** [comment]
Can still reproduce this bug in Godot 3.2.4rc1 (Windows 10 64-bit)

**Comment by akien-mga** [comment]
The debugger issue seems fixed in 3.4, I get the correct error message with Multi-Threaded physics.

The original physics error triggered by the MRP is much harder to reproduce though, had to spam input many times to finally manage to get it to show. I reproduced it a few times and it always showed the proper error message, like in Single-Threaded (where the error happens more easily).
